net/netns.go

```diff
@@ -14,19 +14,7 @@ import (
 
 var ErrLinkNotFound = errors.New("Link not found")
 
-// NB: The following function is unsafe, because:
-//     - It changes a network namespace (netns) of an OS thread which runs
-//       the function. During execution, the Go runtime might clone a new OS thread
-//       for scheduling other go-routines, thus they might end up running in
-//       a "wrong" netns.
-//     - runtime.LockOSThread does not guarantee that a spawned go-routine on
-//       the locked thread will be run by it. Thus, the work function is
-//       not allowed to spawn any go-routine which is dependent on the given netns.
-
-//     Please see https://github.com/weaveworks/weave/issues/2388#issuecomment-228365069
-//     for more details and make sure that you understand the implications before
-//     using the function!
-func WithNetNSUnsafe(ns netns.NsHandle, work func() error) error {
+func WithNetNS(ns netns.NsHandle, work func() error) error {
```

Comment:
Not related to your change, but maybe it makes sense to change `defer netns.Set(oldNs)` bellow to `defer func() { if err := netns.Set(oldNs); err != nil { panic(err) } }()`, because currently any error is ignored?